core/src/avm2/script.rs

```diff
@@ -0,0 +1,286 @@
+//! Whole script representation
+
+use crate::avm1::AvmString;
+use crate::avm2::class::Class;
+use crate::avm2::method::{BytecodeMethod, Method};
+use crate::avm2::r#trait::Trait;
+use crate::avm2::Error;
+use gc_arena::{Collect, Gc, GcCell, MutationContext};
+use std::collections::HashMap;
+use std::mem::drop;
+use std::rc::Rc;
+use swf::avm2::types::{AbcFile, Index, Script as AbcScript};
+
+#[derive(Clone, Debug, Collect)]
+#[collect(require_static)]
+pub struct CollectWrapper<T>(T);
+
+#[derive(Copy, Clone, Debug, Collect)]
+#[collect(no_drop)]
+pub struct TranslationUnit<'gc>(GcCell<'gc, TranslationUnitData<'gc>>);
```

Comment:
So, the name came about while I was switching the AVM2 interpreter away from pulling ABC traits and data directly to storing them in other structures that could be dynamically created (which we need for `avm2-nativeclasses`.) I can't call it `AbcFile` because that's already in use by `swf`, so `TranslationUnit` (a term used by C/C++ compiler specifications to refer to a single source file) seemed like a good option, since, as far as I could tell I was getting one `DoABC` per source file in my tests...

If you have a better option I'll definitely change the name.